### PR TITLE
Fixed as_poly() by making it raise error with unsupported data types eg: tuples

### LINKED ISSUE #18038: What is the result of integrate of a tuple (i.e. integrate((x, x)))?
I just accidently discovered unexpected (at least, for me) behaviour of `integrate` if its first argument is a tuple:

```
In [60]: integrate((x, x))
Out[60]:
 3
x     2
── + x
2

In [61]: integrate((x, x, x))
Out[61]:
 4    3
x    x     2
── + ── + x
3    2

In [62]: integrate((x, 1))
Out[62]:
 3
x
── + x
2
```
It seems to be undocumented and I cannot figure out what does it do. Probably, documentation should be updated to address this.

### PR DESCRIPTION
**Pull request by dhruvmendiratta6**
#### References to other Issues or PRs

Fixes #18038 

#### Brief description of what is fixed or changed

The commit fixes Issue #18038 in which as_poly() would work with tuples and give wrong output. This was fixed by making it an attribute of Expr as tuples are not a subclass Expr. Tests for the same were added  in sympy/core/tests/test_expr.py and sympy/integrals/tests/test_integrals.py . This ultimately fixed the issue where integrate((x, x)) would give a wrong result.

#### Release Notes

* core
    * Moved .as_poly() from Basic to Expr.

### CODE REVIEW

File: sympy/integrals/tests/test_integrals.py [hunk 1]
```diff
@@ -215,6 +215,10 @@ def test_issue_3560():
     assert integrate(1/sqrt(x)**3, x) == -2/sqrt(x)
 
 
+def test_issue_18038():
+    raises(AttributeError, lambda: integrate((x, x))
```
Comment (jksuom) [inline sympy/integrals/tests/test_integrals.py]: One more parenthesis should be appended. Now a syntax error results.

Comment (dhruvmendiratta6) [reply sympy/integrals/tests/test_integrals.py]: I made the changes but Travis is still showing some some error

Comment (jksuom) [reply sympy/integrals/tests/test_integrals.py]: That was an internet connection issue. I have restarted the job.

**Comment by jksuom** [comment]
There is one more change to be made:
```
--- a/doc/src/modules/polys/basics.rst
+++ b/doc/src/modules/polys/basics.rst
@@ -69,7 +69,7 @@ keyword parameter. By default, it is determined by the coefficients
 of the polynomial arguments.
 
 Polynomial expressions can be transformed into polynomials by the
-method :obj:`sympy.core.basic.Basic.as_poly`::
+method :obj:`sympy.core.expr.Expr.as_poly`::
 
     >>> e = (x + y)*(y - 2*z)
     >>> e.as_poly()
```

**Comment by jksuom** [comment]
Thank you for your contribution.

**Pull request merged**